code/multisig/contracts/AAFactory.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import "@matterlabs/zksync-contracts/l2/system-contracts/Constants.sol";
+import "@matterlabs/zksync-contracts/l2/system-contracts/libraries/SystemContractsCaller.sol";
+
+contract AAFactory {
+    bytes32 public aaBytecodeHash;
+
+    constructor(bytes32 _aaBytecodeHash) {
+        aaBytecodeHash = _aaBytecodeHash;
+    }
+
+    function deployAccount(
+        bytes32 salt,
+        address owner1,
+        address owner2
+    ) external returns (address accountAddress) {
+        (bool success, bytes memory returnData) = SystemContractsCaller
+            .systemCallWithReturndata(
+                uint32(gasleft()),
+                address(DEPLOYER_SYSTEM_CONTRACT),
+                uint128(0),
+                abi.encodeCall(
+                    DEPLOYER_SYSTEM_CONTRACT.create2Account,
+                    (salt, aaBytecodeHash, abi.encode(owner1, owner2), IContractDeployer.AccountAbstractionVersion.Version1)
+                )
+            );
+        require(success, "Deployment failed");
+
+        (accountAddress) = abi.decode(returnData, (address));
+    }
+}
```

Comment:
Do you think it would be cleaner to advocate for using `create2Account` from [`system-contracts/contracts/Create2Factory.sol`](https://github.com/matter-labs/era-contracts/blob/bce4b2d0f34bd87f1aaadd291772935afb1c3bd6/system-contracts/contracts/Create2Factory.sol#L27)? I am just thinking out loud here so let me know your thoughts but rather then having to call `SystemContractsCaller` we:

```
import "@matterlabs/zksync-contracts/l2/system-contracts/contracts/Create2Factory.sol";

address deployUsingSystemContractDeployer = AAFactory.create2Account(
       salt,
       bytecodeHash,
      abi.encode(),
      IContractDeployer.AccountAbstractionVersion.Version1
);
```

Perhaps we dont make the change immediately but I think this is a better approach to promote, thoughts?

Reply:
nice that looks a lot cleaner. Any reason why you would want to wait to implement?

Reply:
I dont think `Create2Factory` is included in the published version of era-contracts 👎

Reply:
If it is we should tho.